packages/runtime-core/__tests__/apiInject.spec.ts

```diff
@@ -91,6 +92,34 @@ describe('api: provide/inject', () => {
     expect(serialize(root)).toBe(`<div>foobar</div>`)
   })
 
+  it('bound to instance', () => {
+    const Provider = {
+      setup() {
+        return () => h(Consumer)
+      }
+    }
+
+    const Consumer = defineComponent({
+      name: 'Consumer',
+      inject: {
+        foo: {
+          from: 'foo',
+          default() {
+            return this!.$options.name
+          }
+        }
+      },
+      render() {
+        // @ts-ignore
```

Comment:
I don't know how to make this work without this ts-ignore. Any idea?

Reply:
it's ok for the test case.